api/client/search.go

```diff
@@ -21,12 +23,25 @@ import (
 func (cli *DockerCli) CmdSearch(args ...string) error {
 	cmd := Cli.Subcmd("search", []string{"TERM"}, Cli.DockerCommands["search"].Description, true)
 	noTrunc := cmd.Bool([]string{"-no-trunc"}, false, "Don't truncate output")
-	automated := cmd.Bool([]string{"-automated"}, false, "Only show automated builds")
-	stars := cmd.Uint([]string{"s", "-stars"}, 0, "Only displays with at least x stars")
+	filter := cmd.String([]string{"f", "-filter"}, "", "Use filters (is-automated, is-official, has-stars)")
```

Comment:
We should use the same way of declaring filters as the other command that support filters (like ps — [`api/client/ps.go`](https://github.com/docker/docker/blob/master/api/client/ps.go)) and use `"github.com/docker/engine-api/types/filters"` to parse filter from args.

```go
var filterArgs = filters.NewArgs()
var flFilter = opts.NewListOpts(nil)
cmd.Var(&flFilter, []string{"f", "-filter"}, "Filter output based on conditions provided")
// […]
for _, f := range flFilter.GetAll() {
    if filterArgs, err = filters.ParseFlag(f, filterArgs); err != nil {
        return err
    }
}
// […]
```